www/codePush.ts

```diff
@@ -206,7 +211,58 @@ class CodePush implements CodePushCordovaPlugin {
      * 
      */
     public sync(syncCallback?: SuccessCallback<any>, syncOptions?: SyncOptions, downloadProgress?: SuccessCallback<DownloadProgress>): void {
-
+        /* Check if a sync is already in progress */
+        if (CodePush.SyncInProgress) {
+            /* A sync is already in progress */
+            CodePushUtil.logMessage("Sync already in progress.");
+            syncCallback && syncCallback(SyncStatus.IN_PROGRESS);
+        } else {
+            /* Create a callback that resets the SyncInProgress flag when the sync is complete
+             * If the sync status is a result status, then the sync must be complete and the flag must be updated
+             * Otherwise, do not change the flag and trigger the syncCallback as usual
+             */
+            var syncCallbackAndUpdateSyncInProgress: SuccessCallback<any> = (result: any): void => {
+                switch (result) {
+                    case SyncStatus.ERROR:
+                    case SyncStatus.IN_PROGRESS:
+                    case SyncStatus.UP_TO_DATE:
+                    case SyncStatus.UPDATE_IGNORED:
+                    case SyncStatus.UPDATE_INSTALLED:
+                        /* The sync has completed */
+                        CodePush.SyncInProgress = false;
+                        break;
```

Comment:
i was under the impression that every case needs to exit with a break at some point, so i could take out the intermediate states and only have a break at the end